Gems/AtomLyIntegration/CommonFeatures/Code/Source/Mesh/MeshComponentController.cpp

```diff
@@ -699,6 +701,74 @@ namespace AZ
             return Aabb::CreateNull();
         }
 
+        void MeshComponentController::GetVisibleGeometry(AzFramework::VisibleGeometryContainer& geometryContainer) const
+        {
+            // Attempt to copy the triangle list geometry data out of the model asset into the visible geometry structure
```

Comment:
The geometryContainer should probably get a clear() call at the top here so that even in error conditions, it's guaranteed not to have any stale incorrect data.

Reply:
The container is cleared at the call site and passed by reference in case there are multiple components contributing geometry.

Reply:
@lumberyard-employee-dm  - visibleGeometry is a local that gets added to geometryContainer:
```
                geometryContainer.emplace_back(AZStd::move(visibleGeometry));
```
I was asking about the container itself. For geometryContainer, if the intent is for this method to add to it but not replace the contents, that should get spelled out very explicitly in the comments, along with the fact that clearing the container is the responsibility of the caller. Right now, the comment says `//! Returns a container of visible geometry` which apparently isn't accurate. It's appending to the passed-in container of visible geometry, not returning a new one.